### PR TITLE
doc: Update release process doc to accurately reflect current process

### PR DESCRIPTION
**Pull request by djgalloway**
Signed-off-by: David Galloway <dgallowa@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/dev/release-process.rst [hunk 1]
```diff
@@ -2,172 +2,105 @@
   Ceph Release Process
 ======================
 
-1. Build environment
-====================
+Prequisites
```
Comment (zdover23) [inline doc/dev/release-process.rst:5]: s/Prequisites/Prerequisites/

**Review by zdover23 — CHANGES_REQUESTED** [review]

File: doc/dev/release-process.rst [hunk 2]
```diff
@@ -2,172 +2,105 @@
...
+===========
 
-There are multiple build environments, debian based packages are built via pbuilder for multiple distributions.  The build hosts are listed in the ``deb_hosts`` file, and the list of distributions are in ``deb_dist``.  All distributions are build on each of the build hosts.  Currently there is 1 64 bit and 1 32 bit build host.
+Signing Machine
+---------------
+The signing machine is a virtual machine in the `Sepia lab<https://wiki.sepia.ceph.com/doku.php?id=start>`_.  SSH access is limited to the usual Infrastructure Admins along with a few other component leads (e.g., nfs-ganesha, ceph-iscsi).
```
Comment (anthonyeleven) [inline doc/dev/release-process.rst:10]: Suggest s/virtual machine/VM/ to be less stilted.

**Review by anthonyeleven — CHANGES_REQUESTED** [review]

File: doc/dev/release-process.rst [hunk 3]
```diff
@@ -2,172 +2,105 @@
...
+Prerequisites
+=============
 
-There are multiple build environments, debian based packages are built via pbuilder for multiple distributions.  The build hosts are listed in the ``deb_hosts`` file, and the list of distributions are in ``deb_dist``.  All distributions are build on each of the build hosts.  Currently there is 1 64 bit and 1 32 bit build host.
+Signing Machine
+---------------
+The signing machine is a virtual machine in the `Sepia lab <https://wiki.sepia.ceph.com/doku.php?id=start>`_.  SSH access is limited to the usual Infrastructure Admins along with a few other component leads (e.g., nfs-ganesha, ceph-iscsi).
 
-The RPM based packages are built natively, so one distribution per build host.  The list of hosts is found in ``rpm_hosts``.
+The ``ubuntu`` user on the machine has a few `build scripts` <https://github.com/ceph/ceph-build/tree/main/scripts>`_ to ease the pulling, pushing, and signing of packages.
```
Comment (djgalloway) [inline doc/dev/release-process.rst:12]:
```suggestion
The ``ubuntu`` user on the machine has a few `build scripts <https://github.com/ceph/ceph-build/tree/main/scripts>`_ to ease the pulling, pushing, and signing of packages.
```

File: doc/dev/release-process.rst [hunk 4]
```diff
@@ -2,172 +2,105 @@
...
 
-    export GNUPGHOME=<path to keyring dir>
+Summarized build process
+========================
 
-    # verify it's accessible
-    gpg --list-keys
+1. QE finishes testing and finds a stopping point.  That commit is pushed to the ``$release-release`` branch in ceph.git (e.g., quincy-release).  This allows work to continue in the working ``$release`` branch without having to freeze it during the release process.
+2. The Ceph Council approves and notifies the Build Lead.
+3. Build Lead starts [Jenkins multijob](https://jenkins.ceph.com/view/all/job/ceph) which triggers all builds.
```
Comment (djgalloway) [inline doc/dev/release-process.rst:29]:
```suggestion
3. Build Lead starts `Jenkins multijob <https://jenkins.ceph.com/view/all/job/ceph>`_ which triggers all builds.
```

**Review by zdover23 — APPROVED** [review]

File: doc/dev/release-process.rst [hunk 5]
```diff
@@ -2,172 +2,105 @@
...
-The RPM based packages are built natively, so one distribution per build host.  The list of hosts is found in ``rpm_hosts``.
+The ``ubuntu`` user on the machine has a few `build scripts` <https://github.com/ceph/ceph-build/tree/main/scripts>`_ to ease the pulling, pushing, and signing of packages.
 
-Prior to building, it's necessary to update the pbuilder seed tarballs::
+Finally, the GPG signing key permanently lives on a `Nitrokey Pro <https://shop.nitrokey.com/shop/product/nkpr2-nitrokey-pro-2-3>`_ and is passed through to the VM via RHV.  This helps ensure the key can not be exported or leave the datacenter in any way.
 
-    ./update_all_pbuilders.sh
+New Major Releases
+------------------
+For new major (alphabetical) releases, RPM repos need a `ceph-release` RPM created.  The chacra repos are configured to include this RPM but it must be built separately. You must make sure that chacra is properly configured to include this RPM though for the release you are doing. See `this PR <https://github.com/ceph/chacra/pull/219>`_ for an example.
```
[the remark this answers is not in the document]
Comment (djgalloway) [reply doc/dev/release-process.rst:21]: Correct!

[the remark this answers is not in the document]
Comment (zdover23) [reply doc/dev/release-process.rst:21]: Alright. I'm going to tidy up the syntax here (I expect this to take about another half hour). After that, I think that we can merge this PR.

[the remark this answers is not in the document]
Comment (zdover23) [reply doc/dev/release-process.rst]: I think that the relationship between 1) chacra and 2) the RPM that must be built before chacra is run should be made clearer. 

We could improve this by merely including a sentence explaining what chacra is, and what it does.

[the remark this answers is not in the document]
Comment (zdover23) [reply doc/dev/release-process.rst]: @djgalloway ^^^ (see above)

File: doc/dev/release-process.rst (hunk 2 above)
Comment (anthonyeleven) [reply doc/dev/release-process.rst]: If someone doesn't know what a triode is I don't want them using my software in the first place, but ok.  Maybe instead replace "signing machine" with "signing host"?

**Review by anthonyeleven — APPROVED** [review]

**Comment by zdover23** [comment]
jenkins test docs

**Comment by djgalloway** [comment]
DNM yet.

To-do:

1. Make step 1 pushing to `quincy-release`
2. I'd like to make a table with the distro map (e.g., | quincy | `centos8 centos9 focal bullseye`)

**Comment by zdover23** [comment]
> DNM yet.
> 
> To-do:
> 
>     1. Make step 1 pushing to `quincy-release`
> 
>     2. I'd like to make a table with the distro map (e.g., | quincy | `centos8 centos9 focal bullseye`)
> 
>     3. Edit `sync-pull` command so release name is in < >
> 
>     4. Add note about where packages go under `3. Signing and Publishing the Build`
> 
>     5. Add merfi and sign-rpms output
> 
>     6. Mention the ceph.git release notes pull request under `5. Announce the Release`

@djgalloway Would you like me to take care of these items and push this back to you for review?

Also, there's some kind of error that I've never seen before that is making the docs build check and the ceph-pr-docs check fail. Do you have any insight into that?

**Comment by djgalloway** [comment]
> @djgalloway Would you like me to take care of these items and push this back to you for review?

I appreciate the offer it's all knowledge in my head that needs to get out of my head :rofl:.

> Also, there's some kind of error that I've never seen before that is making the "docs build check" and the "ceph-pr-docs check" fail. Do you have any insight into that?

I haven't looked yet but I'll fix it.

**Comment by djgalloway** [comment]
@zdover23 I think I'm done.  Please take a look at and let me know if

1. You think you could follow these steps and create a Ceph release (if you had proper access to all the systems)
2. Formatting, syntax, grammar, etc. is up to your standards

https://ceph--47736.org.readthedocs.build/en/47736/dev/release-process/

**Comment by zdover23** [comment]
> 

1. I think so. I think that I could use this procedure to create a Ceph release if I had the necessary permissions.
2. I've made an alteration to the formatting of one of the procedures. There are minor usage and syntactical alterations that I might make in the future, but I don't see anything here that is a showstopper--nothing here should stop this merge. This is excellent internal procedural documentation, and if international conventions open again I might use it as the example of how to create such internal documentation.

**Comment by zdover23** [comment]
> @zdover23 I think I'm done. Please take a look at and let me know if
> 
>     1. You think you could follow these steps and create a Ceph release (if you had proper access to all the systems)
> 
>     2. Formatting, syntax, grammar, etc. is up to your standards
> 
> 
> https://ceph--47736.org.readthedocs.build/en/47736/dev/release-process/

1. I think so. I think that I could use this procedure to create a Ceph release if I had the necessary permissions.
2. I've made an alteration to the formatting of one of the procedures. There are minor usage and syntactical alterations that I might make in the future, but I don't see anything here that is a showstopper--nothing here should stop this merge. This is excellent internal procedural documentation, and if international conventions open again I might use it as the example of how to create such internal documentation.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/47837 - Quincy backport
https://github.com/ceph/ceph/pull/47838 - Pacific backport

**Pull request merged**